tower-http/src/services/fs/serve_dir.rs

```diff
@@ -192,6 +203,34 @@ impl<ReqBody> Service<Request<ReqBody>> for ServeDir {
     }
 }
 
+fn parse_range(headers: &HeaderMap) -> Option<ParsedRangeHeader> {
+    // ex:  `Range: bytes=0-1023`
+    let range_header = headers.get(header::RANGE)?;
+    if let Ok(as_str) = range_header.to_str() {
+        // produces ("bytes", "0-1023")
+        return if let Some((_, range)) = as_str.split_once("=") {
+            // produces ("0", "1023")
+            if let Some((start_incl, end_incl)) = range.split_once("-") {
+                if let Ok(start) = start_incl.parse() {
+                    if let Ok(end) = end_incl.parse() {
+                        if start < end {
+                            return Some(ParsedRangeHeader::Valid(ContentRange {
+                                start_inclusive: start,
+                                end_inclusive: end
+                            }));
+                        }
+                    }
+                }
+            }
```

Comment:
Might be good to run some tests with resuming downloads from a browser and see the actual header values that are sent and make sure this handles all of them.

We talked about the many forms the `Range` header can take so might good to make sure we cover the necessary cases.

Reply:
Should probably also have some unit tests specifically for this function.